include/aspect/parameters.h

```diff
@@ -292,6 +292,15 @@ namespace aspect
      */
     unsigned int                   n_compositional_fields;
     std::vector<std::string>       names_of_compositional_fields;
+    std::map<std::string,std::string> advection_methods_of_compositional_fields;
+
+    /**
+     * Map from compositional field name to a pair "particle property", "component",
+     * where component is of the format "[1][2][3]" and the particle property is
```

Comment:
I don't understand ``format "[1][2][3]"`` because you store it in an int? Are you trying to say that the component is a value between 1 and dim?